packages/server/src/main/java/software/uncharted/terarium/hmiserver/service/data/TerariumAssetService.java

```diff
@@ -133,13 +136,34 @@ public void deleteAsset(final UUID id) throws IOException {
 	 * @throws IOException If there is an error creating the asset
 	 */
 	public T createAsset(final T asset) throws IOException {
-		final UUID id = UUID.randomUUID();
-		asset.setId(id);
+		if (elasticService.documentExists(getAssetIndex(), asset.getId().toString())) {
+			throw new ResponseStatusException(
```

Comment:
Because this isn't a controller, do we want to throw a different kind of exception here to use it more generically? (Then let the controller package it nicely?)

Reply:
Done!